c10/core/DispatchKey.h

```diff
@@ -347,11 +347,13 @@ enum class DispatchKey : uint16_t {
   VmapMode,
 
   FuncTorchGradWrapper, // See Note [Out-of-tree vmap+grad prototype]
+
   // Alias and mutation removal.
   // If some backends want to opt into only alias removal or only mutation
   // removal,
   // we can consider adding separate keys dedicated to those individual passes.
   // See Note [Functionalization Pass In Core] for details.
+  FunctionalizeAddBackViews,
```

Comment:
I'm wondering if we actually need a separate key for this. An alternative is to just have some TLS that Functionalize queries when implementing `_copy` operations.

Reply:
Yeah I think I should just do this. We'll save the extra key, and it also feels easier to reason about than the extra redispatch (where in functorch, now you need to worry about having two redispatch's "sandwiched" inside of DynamicLayer).

Current plan: we already have a `FuncTorchTLS` construct in core, and `AddBackViews` is currrently only a useful concept in the context of the `functionalize()` transform, so I'm thinking of adding a flag in there for it.

The functionalize view kernels would just check that TLS right before redispatching (to either a `view` or a `view_copy` op), and functorch's `DynamicLayer` logic would be responsible for setting / un-setting that TLS upon entering / exiting the layer.

(We could probably do something really similar for "preserving mutations", which I remember discussions about that being useful to mobile).